packages/node_modules/@node-red/nodes/core/common/lib/debug/debug-utils.js

```diff
@@ -243,7 +272,18 @@ RED.debug = (function() {
         });
         candidateNodes = candidateNodes.filter(function(node) {
             return workspaceOrderMap.hasOwnProperty(node.z);
-        })
+        });
+        var map = {};
+        RED.nodes.eachNode(function (n) {
```

Comment:
I think this code can be made more efficient.

At the moment it will iterate over *all* nodes at least two times - first on line 267 with the call to `RED.nodes.filterNodes` and again here on line 277 with `RED.nodes.eachNode`.

I *think* that can be simplified to only have to do one loop over all the nodes.

The code on line 273 filters the list of all debug nodes to be just those on normal flows. That loop is also an opportunity to build a map of debug nodes that exist inside subflows. It can also build a list of all subflow instance nodes.

The second iteration can then be reduced to just checking the subflow instance nodes to see if they contain debug nodes.

Reply:
Hi. Rearranged loops to match your suggestion.  
I think it works better for large flows.